<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
 <HEAD>
   <TITLE> [rabbitmq-discuss] Pika disconnects
   </TITLE>
   <LINK REL="Index" HREF="index.html" >
   <LINK REL="made" HREF="mailto:rabbitmq-discuss%40lists.rabbitmq.com?Subject=Re%3A%20%5Brabbitmq-discuss%5D%20Pika%20disconnects&In-Reply-To=%3CCACDr4icoabgjk_Pf1EvpBU-ERyr5WiaS73jSCV_pJvJNBEbcVQ%40mail.gmail.com%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="015550.html">
   <LINK REL="Next"  HREF="015413.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[rabbitmq-discuss] Pika disconnects</H1>
    <B>John Reuning</B> 
    <A HREF="mailto:rabbitmq-discuss%40lists.rabbitmq.com?Subject=Re%3A%20%5Brabbitmq-discuss%5D%20Pika%20disconnects&In-Reply-To=%3CCACDr4icoabgjk_Pf1EvpBU-ERyr5WiaS73jSCV_pJvJNBEbcVQ%40mail.gmail.com%3E"
       TITLE="[rabbitmq-discuss] Pika disconnects">john at ibiblio.org
       </A><BR>
    <I>Fri Oct  7 13:49:51 BST 2011</I>
    <P><UL>
        <LI>Previous message: <A HREF="015550.html">[rabbitmq-discuss] Migration from 2.2.0 to 2.6.1 and cluster
</A></li>
        <LI>Next message: <A HREF="015413.html">[rabbitmq-discuss] Pika disconnects
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#15411">[ date ]</a>
              <a href="thread.html#15411">[ thread ]</a>
              <a href="subject.html#15411">[ subject ]</a>
              <a href="author.html#15411">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>I have a multi-threaded python application using pika to publish
messages to rabbitmq (all centos 6).  It works most of the time,
except for intermittent disconnects and bad file descriptor socket
errors.  The disconnects manifest as the ioloop exiting.  I registered
a custom on_connection_close callback function, as well as a custom
reconnection strategy object.  Neither is triggered when a disconnect
occurs and the ioloop exists.  The socket EBADF exceptions come from
calling basic_publish on a channel object after the rabbitmq
connection has terminated.

I read that pika isn't thread safe, so I've tried adding a
threading.RLock around the basic_publish call.  This reduces the
frequency of the disconnects but doesn't eliminate them.  I could wrap
the pika functionality in another layer, check for a disconnect, and
recycle everything.  However, it would be nice to know why this
happens.  Anyone have suggestions?

Thanks,

-John
</PRE>














<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="015550.html">[rabbitmq-discuss] Migration from 2.2.0 to 2.6.1 and cluster
</A></li>
	<LI>Next message: <A HREF="015413.html">[rabbitmq-discuss] Pika disconnects
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#15411">[ date ]</a>
              <a href="thread.html#15411">[ thread ]</a>
              <a href="subject.html#15411">[ subject ]</a>
              <a href="author.html#15411">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="https://lists.rabbitmq.com/cgi-bin/mailman/listinfo/rabbitmq-discuss">More information about the rabbitmq-discuss
mailing list</a><br>
</body></html>
